src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/PublisherEvent.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Consts.Table;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.PersistableAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.persist.JooqFieldValue;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.PersistableObject;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.PersistorBase;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.TableInfo;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+
+/**
+ * A persistable object that persists all actionable actions published by Publisher
+ * This object will be inserted as a tuple in "PublisherEvent" table. And any actionable actions will
+ * be persisted separately in Action table and linked to this tuple via foreign key.
+ * By joining one PublisherEvent with the Action table, we can map one decision(here it is PublisherEvent)
```

Comment:
I think we only need to persist actions published by publisher. Do you see a need to store all actions that we create irrespective of whether or not they are published?